Engine/src/Platform/OpenGL41/OpenGL41Texture.cpp

```diff
@@ -0,0 +1,105 @@
+#include "OpenGL41Texture.h"
+
+#include "Platform/OpenGL41/OpenGL41Check.h"
+#include "Platform/OpenGLCommon.h"
+
+#include <glad/gl.h>
+
+namespace elv {
+
+OpenGL41Texture::OpenGL41Texture(std::uint32_t width, std::uint32_t height, std::uint32_t nrChannels)
+    : m_width(width)
+    , m_height(height)
+{
+
+    if (nrChannels == 4) {
+        m_internalFormat = GL_RGBA8;
+        m_dataFormat = GL_RGBA;
+    } else if (nrChannels == 3) {
+        m_internalFormat = GL_RGB8;
+        m_dataFormat = GL_RGB;
+    } else {
+        m_internalFormat = GL_R8;
+        m_dataFormat = GL_RED;
+    }
+
+    glCheck(glGenTextures(1, &m_id));
+    glCheck(glBindTexture(GL_TEXTURE_2D, m_id));
+    glTexImage2D(GL_TEXTURE_2D, 0, m_internalFormat, m_width, m_height, 0, m_dataFormat, GL_UNSIGNED_BYTE, nullptr);
+
+    glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT));
+    glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT));
+    glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR));
+    glCheck(glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR));
+    glCheck(glBindTexture(GL_TEXTURE_2D, 0));
+}
+
+OpenGL41Texture::OpenGL41Texture(std::uint32_t width, std::uint32_t height, const Info& info)
+    : m_width(width)
+    , m_height(height)
+{
+    const auto format = OpenGL::GetGLDataFormat(info.InternalFormat);
+
+    m_internalFormat = format.first;
+    m_dataFormat = format.second;
+
+    // if (info.Multisampled) {
+    //     // TODO:  Implement multisampled texture
+    //     //  glCreateTextures(GL_TEXTURE_2D_MULTISAMPLE, 1, &m_id);
+    //     //  glTextureStorage2DMultisample(m_id, 4, format.first, width, height, GL_TRUE);
+    // } else {
+    glCheck(glGenTextures(1, &m_id));
+    glCheck(glBindTexture(GL_TEXTURE_2D, m_id));
+    if (info.InternalFormat == Texture::InternalFormat::DepthStencil) {
```

Comment:
not using GL_UNSIGNED_INT_24_8 specifically for GL_DEPTH_STENCIL texture type was problematic and caused errors, had to add a special branch for that